vumi_http_api/auth.py

```diff
@@ -0,0 +1,59 @@
+from zope.interface import implements
+
+from twisted.cred import portal, checkers, credentials, error
+from twisted.web import resource
+from twisted.web.guard import HTTPAuthSessionWrapper, BasicCredentialFactory
+
+
+class ConversationRealm(object):
+    implements(portal.IRealm)
+
+    def __init__(self, resource):
+        self.resource = resource
+
+    def requestAvatar(self, user, mind, *interfaces):
+        if resource.IResource in interfaces:
+            return (resource.IResource, self.resource, lambda: None)
+        raise NotImplementedError()
+
+
+class ConversationAccessChecker(object):
+    implements(checkers.ICredentialsChecker)
+    credentialInterfaces = (credentials.IUsernamePassword,)
+
+    def __init__(self, worker, conversation_key):
+        self.worker = worker
+        self.conversation_key = conversation_key
+
+    def requestAvatarId(self, credentials):
+        username = credentials.username
+        token = credentials.password
+        tokens = self.worker.get_static_config().api_tokens
+        if token in tokens:
```

Comment:
Maybe we need to have a list of `(account_key, conversation_key, tokens)` in the config, instead of just a list of tokens. That way we can control access to specific conversations for specific accounts.